crates/executor/src/blockchain_tree/block_indices.rs

```diff
@@ -0,0 +1,258 @@
+//! Implementation of [`BlockIndices`] related to [`super::BlockchainTree`]
+
+use super::chain::{Chain, ChainId, ForkBlock};
+use reth_primitives::{BlockHash, BlockNumber, SealedBlockWithSenders};
+use std::collections::{BTreeMap, BTreeSet, HashMap, HashSet};
+
+/// Internal indices of the block.
+#[derive(Clone)]
+pub struct BlockIndices {
+    /// Last finalized block.
+    pub last_finalized_block: BlockNumber,
```

Comment:
I feel like all of the fields should be private.

Reply:
Made them private